File: Java/[2 sem] 2DWorldSimulator/code/Sheep.java
public class Sheep extends Animal {
    private static final int SHEEP_STRENGTH = 4;
    private static final int SHEEP_INITIATIVE = 4;
    private static final char SHEEP_SYMBOL = 'S';

    public Sheep(int x, int y, World world) {
        super(x, y, world, SHEEP_STRENGTH, SHEEP_INITIATIVE, SHEEP_SYMBOL, 0);
    }

    @Override
    public boolean isSheep() {
        return true;
    }

    @Override
    public String getSpecieName() {
        return "Sheep";
    }
}

File: C(++)/[2 sem] 2DWorldSimulator/code/src/worldObjects/animals/Sheep.cpp
//
// Created by szymo on 4/16/2023.
//

#include ".headers/Sheep.h"

#define SHEEP_STRENGTH 4
#define SHEEP_INITIATIVE 4
#define SHEEP_SYMBOL 'S'

Sheep::Sheep(int x, int y, World *world) : Animal(x, y, world, SHEEP_STRENGTH, SHEEP_INITIATIVE, SHEEP_SYMBOL)
{}

bool Sheep::isSheep() { return true; }

std::string Sheep::getSpecieName() const { return "Sheep"; }